### PR TITLE
ROE-809 Validate that the corporate mo start date is in the past

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-809

### Change description

Validate start by checking that it is in the past

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/validation/custom.validation.ts [hunk 1]
```diff
@@ -29,3 +30,13 @@ export const checkDate = (errMsg: string, day: string = "", month: string = "",
   }
   return true;
 };
+
+export const checkDateIsNotFuture = (errMsg: string, day: string = "", month: string = "", year: string = "") => {
```
Comment (markpit) [inline src/validation/custom.validation.ts:34]: I'd change the function name to checkDateIsInPast as today's date will throw an error message even though it is not in the future.

File: src/validation/error.messages.ts [hunk 2]
```diff
@@ -79,5 +79,6 @@ export enum ErrorMessages {
   OCCUPATION_INVALID_CHARACTERS = "Occupation must only include letters a to z, numbers, and special characters such as hyphens, spaces and apostrophes",
   ROLES_AND_RESPONSIBILITIES_INVALID_CHARACTERS = "Role and responsibilities must only include letters a to z, numbers, and special characters such as hyphens, spaces and apostrophes",
   CONTACT_NAME_INVALID_CHARACTERS = "Contact name must only include letters a to z, numbers, and special characters such as hyphens, spaces and apostrophes",
-  EMAIL_INVALID_FORMAT = "Enter an email address in the correct format, like name@example.com"
+  EMAIL_INVALID_FORMAT = "Enter an email address in the correct format, like name@example.com",
+  FUTURE_DATE = "The date it became a managing officer must be in the past"
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/validation/error.messages.ts]: As the beneficial owner other has a similar field, I'd advise keeping the name as generic as possible.

[the remark this answers is not in the document]
Comment (markpit) [reply src/validation/error.messages.ts]: sorry matt, i edited my comment, not sure if you got the update

[the remark this answers is not in the document]
Comment (mattch1) [reply src/validation/error.messages.ts]: Your original idea was better: DATE_NOT_IN_PAST

[the remark this answers is not in the document]
Comment (markpit) [reply src/validation/error.messages.ts]: i had to change it as the message is not generic but the name DATE_NOT_IN_PAST is. MO_START_DATE_NOT_IN_PAST is in keeping with the other messages and describes what the message is for.

[the remark this answers is not in the document]
Comment (mattch1) [reply src/validation/error.messages.ts]: I see what you're saying. Will change this.

**Pull request merged**